### PR TITLE
chore: add devcontainer config

### PR DESCRIPTION
**Pull request by johnstcn**
For convenience!

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "envbuilder",
```
Comment (mafredri) [inline .devcontainer/devcontainer.json:2]: Was this supposed to be something else?

Comment (johnstcn) [reply .devcontainer/devcontainer.json:2]: copy-pasta 😅

**Review by mafredri — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "terraform-provider-coder",
+  "image": "mcr.microsoft.com/devcontainers/go:1.20",
```
Comment (mafredri) [inline .devcontainer/devcontainer.json:3]: Do we still use Go 1.20 here, btw? Should we match coder/coder or go 1.22? 🤔

Comment (johnstcn) [reply .devcontainer/devcontainer.json:3]: https://github.com/coder/terraform-provider-coder/issues/222

**Pull request merged**